scripts/core-triage/project.py

```diff
@@ -0,0 +1,195 @@
+    "help_wanted",
+    "spike",
+    "python_models",
+    "Team:Language",
+    "Team:Execution",
+    "Team:Adapters",
+]
+# prs added by label individually
+PR_LABELS = ["ready_for_review"]
+# GitHub TOKEN environment variable name
+TOKEN_VAR = "GH_TOKEN"
+# Number of items per query -- max is 100 TODO: paging?
+NUM_ITEMS = 100
+
+# API stuff -- not uppercase because...
+headers = {"Authorization": f"token {os.environ.get(TOKEN_VAR)}"}
+gh_api_url = "https://api.github.com"
+gh_graphql_url = f"{gh_api_url}/graphql"
+
+# functions
+def process_request(url: str, headers: dict, json: dict = None) -> dict:
+    """
+    Process a request to the GitHub API (GET or POST, based on json).
+    ---
+    Inputs: url, headers, json (optional)
+    Outputs: response (dict)
+    """
+    if json:
+        r = requests.post(url, headers=headers, json=json)
+    else:
+        r = requests.get(url, headers=headers)
+
+    print(r.status_code)
+    print(r.text)
+
+    return r.json()
+
+
+def get_core_members(teams: list[str]) -> list:
+    """
+    Get a list of Core members' GitHub logins (usernames).
+    ---
+    Inputs: teams (list of str)
+    Outputs: core_members (list of str)
+    """
+    # initialize empty list
+    core_members = []
+
+    # for each core team, add its members
+    for team in teams:
+        team_member_url = f"{gh_api_url}/orgs/{ORG}/teams/{team}/members"
+        r = process_request(team_member_url, headers=headers)
+
+        team_members = [login["login"] for login in r]
+        core_members.extend(team_members)
+
+    return sorted(list(set(core_members)))
+
+
+def get_project_id(project_num: int) -> str:
+    """
+    Get a project's ID from its number.
+    ---
+    Inputs: project_num (int)
+    Outputs: project_id (str)
+    """
+    project_id = process_request(
+        gh_graphql_url,
+        headers=headers,
+        json={
+            "query": project_id_query.replace("$org", f'"{ORG}"').replace(
+                "$project_num", f"{project_num}"
+            )
+        },
+    )["data"]["organization"]["projectV2"]["id"]
+    return project_id
+
+
+def get_issues(repo: str, label: str, num_items: int) -> list[dict]:
+    """
+    Get a list of issues for the given repo and label.
+    ---
+    Inputs: repo (str), label (str), num_items (int)
+    Outputs: issues (list of dict)
+    """
+    # replace variables in query string and filter down to the issues' edges
+    issues = process_request(
+        gh_graphql_url,
+        headers=headers,
+        json={
+            "query": issues_query.replace("$org", f'"{ORG}"')
+            .replace("$repo", f'"{repo}"')
+            .replace("$num_items", f"{num_items}")
+            .replace("$label", f'"{label}"')
+        },
+    )["data"]["repository"]["issues"]["edges"]
+    return issues
+
+
+def add_items_to_project(project_id: str, items: list[dict]) -> None:
+    """
+    Adds items to a project by project_id and a list of item edges.
```

Comment:
Can you call out here that "items" are issues and PRs?

Reply:
done